### PR TITLE
Support for redacting conversation sources and conversation parts

### PR DESCRIPTION
**Pull request by eXeDK**
#### Why?
I needed support for redacting sources and conversations parts and didn't want to use cURL

#### How?
Tried to follow the pattern for implementing calls and tests

#### Please notice
The redact endpoint returns a HTTP 400 error if you try to redact something that is already redacted. That error is not handled here since 400 errors are not handled. Please advise how to catch that efficiently or consider introducing idempotency for the redact endpoint.

### CODE REVIEW

File: lib/intercom/service/conversation.rb [hunk 1]
```diff
@@ -61,6 +61,28 @@ def run_assignment_rules(id)
...
+          source_id = redact_data.fetch(:source_id) { raise 'source_id field is required for type "source"' }
+          params[:source_id] = source_id
+        elsif redact_type == 'conversation_part'
+          conversation_part_id = redact_data.fetch(:conversation_part_id) { raise 'conversation_part_id field is required for type "conversation_part"' }
+          params[:conversation_part_id] = conversation_part_id
+        else
+          raise 'unknown redact_type'
+        end
+
+        response = @client.post("/#{collection_name}/redact", params)
```
Comment (rahulgdsouza) [inline lib/intercom/service/conversation.rb:83]: Like the suggestion of introducing some idempotency in the API.

Meanwhile, we can handle the error here by adding `conversation_part_not_redactable` to L195
https://github.com/intercom/intercom-ruby/blob/1aa86b04698a497aeea2f580c3ed1a9fa3539b1d/lib/intercom/request.rb#L195-L196

Should result in a,
```
Intercom::BadRequestError: Conversation part can't be redacted

**Review by rahulgdsouza — COMMENTED** [review]
@eXeDK 👋  Bit late, but added some feedback, Happy to take this up in a separate PR.
We're looking at making a release `v4.1.2`, can add redaction to it.

File: lib/intercom/service/conversation.rb (hunk 1 above)
Comment (eXeDK) [reply lib/intercom/service/conversation.rb:83]: Can you share some info why some parts cannot be redacted? I've redacted a lot of conversations parts and it seems kind of random being both sources and conversation parts. And how to solve it?

Comment (rahulgdsouza) [reply lib/intercom/service/conversation.rb:83]: As far as I know,
- We've got some info in the docs [here](https://developers.intercom.com/intercom-api-reference/reference#redact-a-conversation-part) outlining which conversation parts can be redacted.
- Then there's the case of trying to redact an already redacted source/part which would result in such an error. 
- And source messages started via articles are  currently not redactable.

Will check with the team and see if there are any more cases, and update the docs too to reflect the same.

**Comment by SeanHealy33** [comment]
Closed due to inactivity.

**Pull request closed without merging**